include/mqtt/endpoint.hpp

```diff
@@ -12030,7 +12028,7 @@ class endpoint : public std::enable_shared_from_this<endpoint<Mutex, LockGuard,
             () mutable {
                 if (!connected_) {
                     // offline async publish is successfully finished, because there's nothing to do.
-                    if (func) func(boost::system::errc::make_error_code(boost::system::errc::success));
+                    if (func) func({});
```

Comment:
The same as the comment above. Explicit error code is easier to understand.

Reply:
Reverted